plugins/api-docs/src/components/ApiDefinitionDialog/ApiDefinitionDialog.tsx

```diff
@@ -0,0 +1,189 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiEntity } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import { BackstageTheme } from '@backstage/theme';
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  makeStyles,
+  Tab,
+  Tabs,
+  Typography,
+} from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+import React, { useEffect } from 'react';
+import { apiDocsConfigRef } from '../../config';
+import { PlainApiDefinitionWidget } from '../PlainApiDefinitionWidget';
+
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  fullHeightDialog: {
+    height: 'calc(100% - 64px)',
+  },
+  root: {
+    display: 'flex',
+    flexGrow: 1,
+    width: '100%',
+    height: '100%',
+    backgroundColor: theme.palette.background.paper,
+  },
+  tabs: {
+    borderRight: `1px solid ${theme.palette.divider}`,
+    flexShrink: 0,
+  },
+  tabContents: {
+    flexGrow: 1,
+    overflowX: 'auto',
+  },
+  title: {
+    color: theme.palette.text.primary,
+    wordBreak: 'break-word',
+    fontSize: theme.typography.h3.fontSize,
+    marginBottom: 0,
+  },
+  type: {
+    textTransform: 'uppercase',
+    fontSize: 11,
+    opacity: 0.8,
+    marginBottom: theme.spacing(1),
+    color: theme.palette.text.primary,
+  },
+}));
+
+function TabPanel(props: {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+}) {
+  const { children, value, index, ...other } = props;
+  const classes = useStyles();
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`vertical-tabpanel-${index}`}
+      aria-labelledby={`vertical-tab-${index}`}
+      className={classes.tabContents}
+      {...other}
+    >
+      {value === index && (
+        <Box pl={3} pr={3}>
+          {children}
+        </Box>
+      )}
+    </div>
+  );
+}
+
+function a11yProps(index: number) {
+  return {
+    id: `vertical-tab-${index}`,
+    'aria-controls': `vertical-tabpanel-${index}`,
+  };
+}
+
+/**
+ * A dialog that lets users inspect the API definition.
+ *
+ * @public
+ */
+export function ApiDefinitionDialog({
+  open,
+  entity,
+  onClose,
+}: {
+  open: boolean;
+  entity: ApiEntity;
+  onClose: () => void;
+}) {
+  const classes = useStyles();
+  const [activeTab, setActiveTab] = React.useState(0);
+
+  useEffect(() => {
+    setActiveTab(0);
+  }, [open]);
+
+  const config = useApi(apiDocsConfigRef);
+  const { getApiDefinitionWidget } = config;
+
+  if (!entity) {
```

Comment:
this should in theory not be possible given the typescript contract, but if we keep this i think it's fine to just return null because it's such a fringe should-be-impossible case